lib/decompress/zstd_decompress_internal.h

```diff
@@ -123,6 +123,9 @@ struct ZSTD_DCtx_s
     const ZSTD_DDict* ddict;     /* set by ZSTD_initDStream_usingDDict(), or ZSTD_DCtx_refDDict() */
     U32 dictID;
     int ddictIsCold;             /* if == 1 : dictionary is "new" for working context, and presumed "cold" (not in cpu cache) */
+    int dictUsesRemaining;       /* if == 1 : dictionary should be used once.
+                                  * if == 0 : dictionary should be forgotten now.
+                                  * if < 0 : dictionary should be used indefinitely. */
```

Comment:
minor : while it works, it nonetheless feels strange to declare a member `dictUsesRemaining` to actually mean "use indefinitely when value is negative".

Would it be clearer to have an `enum`, with typically 3 values, for example `use_indefinitely`, `use_once`, `do_no_longer_use` ?